clash-cores/test/Test/Cores/I2C.hs

```diff
@@ -23,15 +23,17 @@ system0 clk arst = bundle (regFile,done,fault)
   scl  = fmap bitCoerce sclOen
   i2cI = bundle (scl,sdaS)
 
-  (sdaS,regFile) = unbundle
+  (sdaS,registerFile) = unbundle
     (i2cSlave clk (bundle (scl, bitCoerce <$> sdaOen)))
 
   rst = liftA2 (<) rstCounter 500
   rstCounter = register clk arst enableGen (0 :: Unsigned 18) (rstCounter + 1)
 -- See: https://github.com/clash-lang/clash-compiler/pull/2511
-{-# CLASH_OPAQUE system0 #-}
+{-# CLASH_OPAQUE testBench0 #-}
 
-{-# ANN system Synthesize { t_name = "system", t_inputs = [], t_output = PortName "" } #-}
-system = system0 systemClockGen resetGen
+{-# ANN testBench Synthesize { t_name = "testBench", t_inputs = [], t_output = PortName "" } #-}
```

Comment:
As discussed, some comments:
1. `testBench` is a magic name; I think it's better to give it a different name. The `Synthesize` annotation probably masks the special machinery.
2. The record syntax with a space between the constructor and the `{` is confusing, it looks like function application. How many arguments does `ANN` take? :-)
3. We didn't document this yet, but we did discuss steering people away from the constructor and to using `defSyn` instead. I think mainly so we can add fields without existing use sites suddenly having undefined field contents.

Reply:
Ah forget about the `defSyn` that doesn't work... \<sad trombone\>